tests/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixerTest.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Phpdoc;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Phpdoc\NoSuperfluousPhpdocTagsFixer
+ */
+final class NoSuperfluousPhpdocTagsFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        return [
```

Comment:
Can you add cases for:

- Namespaced + imported classed:
```php
use Foo\Bar;

/**
 * @param Bar
 */
function foo(Bar $bar) {}
```
- Namespaced + not imported classes:
```php
/**
 * @param \Foo\Bar
 */
function foo(\Foo\Bar $bar) {}
```
```php
/**
 * @param Foo\Bar
 */
function foo(Foo\Bar $bar) {}
```
- Mix of imported and not imported:
```php
use Foo\Bar;

/**
 * @param \Foo\Bar
 */
function foo(Bar $bar) {}
```
- Imported aliased classes:
```php
use Foo\Bar as Baz;

/**
 * @param Foo\Bar
 */
function foo(Baz $bar) {}
```

Reply:
Thank you, I added the tests and updated the code as most were failing.